src/bootstrap/compile.rs

```diff
@@ -294,9 +298,7 @@ fn compiler_file(compiler: &Path, file: &str) -> PathBuf {
 }
 
 pub fn create_sysroot(build: &Build, compiler: &Compiler) {
-    let sysroot = build.sysroot(compiler);
-    let _ = fs::remove_dir_all(&sysroot);
```

Comment:
This is actually a somewhat crucial step as it cleans out any old artifacts to make sure we don't preserve anything by accident. Could the `*.o` files above be generated into something like `build/$target/rtstartup/foo.o` and then copied from there to the sysroot?